src/main/java/com/dabsquared/gitlabjenkins/GitLabPushTrigger.java

```diff
@@ -379,14 +396,25 @@ private void setBuildCauseInJob(Run run){
         }
     }
 
-    public void onCompleted(Run build){
-        Cause mCause= build.getCause(GitLabMergeCause.class);
+    public void onCompleted(Run run){
+        Cause mCause= run.getCause(GitLabMergeCause.class);
         if (mCause != null && mCause instanceof GitLabMergeCause) {
-            onCompleteMergeRequest(build,(GitLabMergeCause) mCause);
+            onCompleteMergeRequest(run, (GitLabMergeCause) mCause);
+        }
+
+        Cause pCause= run.getCause(GitLabPushCause.class);
+        if (pCause != null && pCause instanceof GitLabPushCause) {
+            onCompletedPushRequest(run, (GitLabPushCause) pCause);
         }
 
     }
 
+    private void onCompletedPushRequest(Run run, GitLabPushCause cause) {
+        if(addCiMessage) {
+            cause.getPushRequest().createCommitStatus(this.getDescriptor().getGitlab().instance(), run.getResult()==Result.SUCCESS?"success":"failure", Jenkins.getInstance().getRootUrl() + run.getUrl());
```

Comment:
According to the [commit status api documentation](http://doc.gitlab.com/ce/api/commits.html#post-the-status-to-commit) the status must be one of the following values:
> state (required) - The state of the status. Can be: pending, running, success, failed, canceled
`"failure"` should be `"failed"`